File: coco/src/main/java/coco/ep/i1/Sub.java
package coco.ep.i1;

import coco.ep.Exp;

public interface Sub<FT> extends coco.ep.m2.Sub<FT>, coco.ep.i1.Exp<FT> {

    default Exp<FT> multby(Exp<FT> other) {
        return this.sub(convert(this.getLeft()).multby(convert(other)), convert(this.getRight()).multby(convert(other)));
    }
}

File: coco_csharp/CoCo/EP/I1/Sub.cs
namespace EP.I1 {
    public interface Sub<FT> : Exp<FT>, EP.M2.Sub<FT> {
        EP.Exp<FT> Exp<FT>.multby(EP.Exp<FT> other) {
            return sub(convert(getLeft()).multby(convert(other)), convert(getRight()).multby(convert(other)));
        }
    }
}